src/service.py

```diff
@@ -47,7 +49,7 @@ def configure(self, snap_config: dict[str, Any]) -> None:
         # Bail out or it will crash on self.snap_client.set() method
         if not snap_config:
             logger.warning("empty snap config: %s, skipping...", snap_config)
-            return
+            raise SnapConfigurationMissingError
```

Comment:
Is this caught anywhere?  How is this better than `snap_client.set()` crashing?

Reply:
If we really want to raise exception,can we use the one raised by `snap`?

```
self.snap_client.set(snap_config, typed=True)
```

Reply:
Usually I prefer self-defined error because the message or naming is more easy to understand. I mean if you compare `SnapError` vs `SnapConfigurationMissingError`, `SnapConfigurationMissingError` give you more information.

Reply:
How's that give you more information, while the error raised by `snap` provide the reason why it failed e.g.`snap.SnapError: Snap: 'rustup'; command ['snap', 'set', 'rustup', '-t'] failed with output = ''
`

Reply:
In addition to this, if we raise here, we should also catch in `charm.py` as @samuelallan72 mentioned

Reply:
It's different, one is empty, another is wrong configuration.

Reply:
> It's different, one is empty, another is wrong configuration.

That's the internal error (actual error) because `snap_client.set` simply run `subprocess` under the hook. So if you pass a `{}` to `snap_client.set`  you are actually running `snap set openstack-exporter -t `, and that's missing an argument

Reply:
Updated